packages/babel-types/scripts/generators/generateBuilders.js

```diff
@@ -11,23 +11,38 @@ module.exports = function generateBuilders() {
  */
 import builder from "../builder";\n\n`;
 
+  const reservedNames = new Set(["super", "import"]);
   Object.keys(definitions.BUILDER_KEYS).forEach(type => {
-    output += `export function ${type}(...args: Array<any>): Object { return builder("${type}", ...args); }
-export { ${type} as ${formatBuilderName(type)} };\n`;
+    const formatedBuilderName = formatBuilderName(type);
+    const formatedBuilderNameLocal = reservedNames.has(formatedBuilderName)
+      ? `_${formatedBuilderName}`
+      : formatedBuilderName;
+    output += `${
+      formatedBuilderNameLocal === formatedBuilderName ? "export " : ""
+    }function ${formatedBuilderNameLocal}(...args: Array<any>): Object { return builder("${type}", ...args); }\n`;
+    // This is needed for backwards compatibility.
+    // It should be removed in the next major version.
+    // arrayExpression -> ArrayExpression
+    output += `export { ${formatedBuilderNameLocal} as ${type} };\n`;
+    if (formatedBuilderNameLocal !== formatedBuilderName) {
+      output += `export { ${formatedBuilderNameLocal} as ${formatedBuilderName} };\n`;
+    }
 
     // This is needed for backwards compatibility.
     // It should be removed in the next major version.
     // JSXIdentifier -> jSXIdentifier
     if (/^[A-Z]{2}/.test(type)) {
-      output += `export { ${type} as ${lowerFirst(type)} }\n`;
+      output += `export { ${formatedBuilderNameLocal} as ${lowerFirst(
+        type
+      )} }\n`;
     }
   });
 
   Object.keys(definitions.DEPRECATED_KEYS).forEach(type => {
     const newType = definitions.DEPRECATED_KEYS[type];
     output += `export function ${type}(...args: Array<any>): Object {
   console.trace("The node type ${type} has been renamed to ${newType}");
-  return ${type}("${type}", ...args);
+  return builder("${type}", ...args);
```

Comment:
here was a bug, which would result in code like this:
```
export function RestProperty(...args: Array<any>): Object {	
  console.trace("The node type RestProperty has been renamed to RestElement");
  return RestProperty("RestProperty", ...args);
}
```
which will cause infinite recursion when using such deprecated builder…